### ISSUE TITLE
Named capture groups in regex literals

### ISSUE THREAD

**Issue by dnalot** [slevithan/xregexp#120]
It'd be nice to be able to pass named capture groups in regex literals to avoid backslash escapes. Would it be feasible to add a syntax like:
```js
XRegexp(/({{name}}\d+)/)
```
Equivalent to:
```js
XRegexp("(?<name>\\d+)")
```

**Comment by dnalot** [comment]
Sorry found it in the [`build` plugin](http://xregexp.com/api/#build)!

**Comment by dnalot** [comment]
After using `build` it'd still be nicer for it to assume `{{name}}` is `?<name>` unless provided?

**Comment by dnalot** [comment]
The following throws an error:
```js
XRegExp.addToken(namedCaptures, ([, name]) => `?<${name}>`)
```
But the equivalent `String.prototype.replace` works?

**Comment by slevithan** [comment]
@dnalot I'm not sure what the issue is you're describing with `XRegExp.addToken`. Can you clarify your problem and the question? Also, what does your `namedCaptures` regex look like?

Note that the output of a an `addToken` handler function needs to be valid ES regex syntax, which `(?<` is not. You can, however, use the `reparse` option (see the docs) if your goal is to output XRegExp syntax that XRegExp then processes for you.

**Comment by dnalot** [comment]
The gist of the original issue is I'm looking for a way to support named regexes in a RegExp literal for convenience. It's especially convenient for interop with CoffeeScript which supports free spacing in `RegExp`s. I tried checking `r.source` for a pattern like `({{name}})` and creating a new `XRegExp` from it. That worked but kinda special case. I tried `addToken` but as clarified it doesn't work for that. I fell back to using `XRegExp.build` but it's not nearly as convenient. The issue linked above describes the context. Feel free to close if that would be undesirable.
